WooCommerce/Classes/Tools/Notices/DefaultNoticePresenter.swift

```diff
@@ -191,6 +193,7 @@ private extension DefaultNoticePresenter {
     func dismiss() {
         noticeOnScreen = nil
         keyboardFrameObserver = nil
+        kvoToken = nil
```

Comment:
Is this sufficient for removing the KVO subscription? It's been a long time since I've done it but I remember having to remove it at `dealloc` (in the old days) as well, perhaps adding it to `deinit` is also required?

Not sure. I can see it getting called even when the notice gets hidden, so perhaps it's fine.

Reply:
Good question, from iOS 11 onwards we do not need to do that, as there's no penalty for letting the observer die before the observed or vice-versa. See [this doc](https://developer.apple.com/library/archive/releasenotes/Foundation/RN-Foundation/index.html) under Relaxed Key-Value Observing Unregistration Requirements for further details, or see this [interesting discussion about it](https://stackoverflow.com/questions/46591637/in-swift-4-how-do-i-remove-a-block-based-kvo-observer).